src/core/NotePlayHandle.cpp

```diff
@@ -196,6 +196,25 @@ void NotePlayHandle::play( sampleFrame * _working_buffer )
 
 	lock();
 
+	// Don't play the note if it falls outside of the user defined key range
+	// TODO: handle the range check by Microtuner, and if the key becomes "not mapped", save the current frequency
+	// so that the note release can finish playing using a valid frequency instead of a 1 Hz placeholder
+	if (key() < m_instrumentTrack->m_firstKeyModel.value() ||
+		key() > m_instrumentTrack->m_lastKeyModel.value())
+	{
+		// Release the note if it started playing before going out of range
+		if (!m_released || m_hasMidiNote)
+		{
+			noteOff(0);
+		}
```

Comment:
If `m_released` is `true`, `noteOff` returns immediately, so this test can be removed and `noteOff` called unconditionally.